src/main/java/build/buildfarm/common/redis/RedisClient.java

```diff
@@ -109,8 +155,57 @@ public <T> T blockingCall(JedisInterruptibleContext<T> withJedis)
     return result;
   }
 
-  @SuppressWarnings("ConstantConditions")
   public <T> T call(JedisContext<T> withJedis) throws IOException {
+    return callImpl(withJedis);
+  }
+
+  private <T> T callImpl(JedisContext<T> withJedis) throws IOException {
+    // Capture all redis problems at the client level.
+    // Try to re-establish the client and log all issues.
+    // This will block the overall thread until redis can be connected to.
+    // It may be a useful strategy for gaining stability on a poorly performing network,
+    // or a redis cluster that goes down.  For example, if you need to perform redis maintance on a
+    // production cluster and do not want to affect build clients.
+    for (int i = 0; i < reconnectClientAttempts; ++i) {
+      try {
+        return defaultCall(withJedis);
+      } catch (Exception redisException) {
+        // Record redis failure.
+        redisErrorCounter.inc();
+        log.log(
```

Comment:
Minor: do we want SEVERE here or ERROR? I wasn’t sure if the semantics of SEVERE weren’t recoverable.

Reply:
switched to ERROR

Reply:
Oh there is no `Level.ERROR`! 
switched to`Level.WARNING`.